### DISCUSSION TITLE
Support option to impersonate account

### DISCUSSION THREAD

**Discussion by marian2js** [wevm/wagmi#1256] [Ideas]
This proposal is to add the `impersonate` option to `createClient`. When this is set to a valid address, then all the instances of `useAccount` will return the impersonated address.

This feature is useful for debugging customer issues. When something doesn't work with a given account, devs can just impersonate the account and see the error by themselves.

A second use case is to allow a user to see the app as any other account, so they can follow up on the state of a cold wallet. All the info is already public on the blockchain, so it doesn't add any privacy concerns.

**Comment by swheel33** [comment]
I've found this extension very useful for debugging: https://chrome.google.com/webstore/detail/impersonator/hgihfkmoibhccfdohjdbklmmcknjjmgl
